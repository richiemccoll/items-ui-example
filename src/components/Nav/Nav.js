import React from "react";
import { Button } from "semantic-ui-react";
import PropTypes from 'prop-types';
import styled from "styled-components";
import Search from "../Search/Search";

const Container = styled.ul`
    display: flex;
    flex-direction: column;
`;

const Item = styled.li`
    margin: .75em;
    list-style: none;
`;

const Nav = ({ toggleForm, showingForm, handleSearch }) => {
  return (
    <nav>
      <Container>
        {!showingForm && <Item><Search handleSearch={handleSearch} /></Item>}
        <Item>
          <Button basic color="blue" onClick={() => toggleForm("add")}>
            {showingForm ? "Back" : "Add new cake"}
          </Button>
        </Item>
      </Container>
    </nav>
  );
};

export default Nav;

Nav.PropTypes = {
  toggleForm: PropTypes.func,
  showingForm: PropTypes.bool,
  handleSearch: PropTypes.func
}
