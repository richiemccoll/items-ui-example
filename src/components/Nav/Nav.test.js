import React from "react";
import Nav from "./Nav";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import sinon from "sinon";
import chai from "chai";

describe("Nav Component", () => {
  const navTest = {
    showingForm: false,
    toggleForm() {
      return;
    },
    handleSearch() {
      return;
    }
  };

  it("should render correctly based on props", () => {
    const component = renderer.create(
      <Nav
        toggleForm={navTest.toggleForm}
        showingForm={navTest.showingForm}
        handleSearch={navTest.handleSearch}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should not show search if form is active", () => {
    const wrapper = mount(<Nav showingForm="true" />);
    chai.expect(wrapper.find("#search").length).to.equal(0);
  });

  it("should show the search if form is not active", () => {
    const wrapper = mount(<Nav showingForm="false" />);
    chai.expect(wrapper.find("#search").length).to.equal(0);
  });

  it("should call toggleForm", () => {
    const toggleForm = sinon.spy();
    const wrapper = mount(
      <Nav toggleForm={toggleForm} showingForm={navTest.showingForm} />
    );
    wrapper.find("button").simulate("click");
    chai.expect(toggleForm).to.have.property("callCount", 1);
  });
});
