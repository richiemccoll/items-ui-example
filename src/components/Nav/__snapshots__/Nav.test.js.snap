// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component should render correctly based on props 1`] = `
<nav>
  <ul
    className="sc-bwzfXH bRskXu"
  >
    <li
      className="sc-htpNat bEgqAd"
    >
      <input
        className="sc-bdVaJa dsiGYE"
        id="search"
        onChange={[Function]}
        placeholder="Search"
      />
    </li>
    <li
      className="sc-htpNat bEgqAd"
    >
      <button
        className="ui blue basic button"
        onClick={[Function]}
        tabIndex={undefined}
      >
        Add new cake
      </button>
    </li>
  </ul>
</nav>
`;
